### PR TITLE
usnic: Keep progress thread alive for CQs.

### PR DESCRIPTION
**Pull request by bturrubiates**
A common usage pattern for external FD use is to acquire the FD, call
trywait, and call fi_cq_read. The progress threads needs to stay alive
during the duration of the wait in order to progress receives. Keep
track of CQs that are waiting and increment the fabric wait var to keep
the thread alive. Decrement once again upon the first fi_cq_read for
each CQ.

@goodell @jsquyres 

Signed-off-by: Ben Turrubiates <bturrubi@cisco.com>

### CODE REVIEW

File: prov/usnic/src/usdf.h [hunk 1]
```diff
@@ -376,6 +376,7 @@ struct usdf_cq {
 	struct usdf_domain *cq_domain;
 	struct fi_cq_attr cq_attr;
 	uint8_t is_soft;
+	uint8_t waiting;
```
Comment (goodell) [inline prov/usnic/src/usdf.h]: Nit: these two fields don't match the naming prefix scheme (`cq_`).

Comment (bturrubiates) [inline prov/usnic/src/usdf.h]: I could fix waiting here, and submit a separate PR to update `is_soft` since it isn't really related to this one. That fine?

Comment (goodell) [inline prov/usnic/src/usdf.h]: works for me

**Comment by goodell** [comment]
other than the nit, :+1:
